daemon/policy.go

```diff
@@ -330,7 +414,10 @@ func (d *Daemon) PolicyDelete(labels labels.LabelArray) (uint64, error) {
 	// Stop polling for ToFQDN DNS names for these rules
 	d.dnsRuleGen.StopManageDNSName(rules)
 
-	d.TriggerPolicyUpdates(false, "policy rules deleted")
+	// Only regenerate endpoints which are needed to be regenerated as a result
+	// of the rule update. The rules which were imported most likely do not select
+	// all endpoints in the policy repository (and may not select any at all).
+	go d.ReactToRuleUpdates(&policySelectionWG, epsIdentityConsumers, endpointsToRegen, rev)
```

Comment:
no need for go routine here as this function will also initialize a bunch of go routines

Reply:
It is necessary, because `ReactToRuleUpdates` waits on the provided `sync.WaitGroup`.